ag/validation.go

```diff
@@ -85,6 +82,23 @@ func Interceptor(logger *zap.Logger, userMap map[string]uint64) grpc.UnaryServer
 	}
 }
 
+// Returns modified metadata containing a valid user. Returns an error if the user is not authenticated.
+func UserValidation(meta metadata.MD, userMap map[string]uint64) (metadata.MD, error) {
```

Comment:
Maybe this function doesn't need to be exported??

If the export is useful, then the comment should be phrased like this:
```golang
// UserValidation returns modified ... An error is returned if the user ...
```

This is the recommended Go-style for comments, to always begin with the function name.